### PR TITLE
Added better tests for math ops and unified them

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #21131 Added degrees, radians, ldexp
* #21130 Added frexp, isinf, isnan, isfinite
* #21129 Added factorial
* #21128 Finished trigonometric functions
* #21127 Finished the high-priority functions
* #21126 Added remainder of high-priority trigonometric math ops
* **#21125 Added better tests for math ops and unified them**

Differential Revision: [D15563188](https://our.internmc.facebook.com/intern/diff/D15563188)

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -5952,84 +5952,86 @@ def func(a, b):
...
+                except Exception as e:
+                    resf = e
+                try:
+                    resfs = fs(a, b)
+                except Exception as e:
+                    resfs = e
+                if debug:
+                    print("in: ", a, b)
+                    print("out: ", resf, resfs)
+                if resf != resfs:
```
Comment (Chillee) [inline test/test_jit.py]: I can't use `assertEqual` here because there are some specific conditions (`nan` should be equal, we want to silently ignore errors, etc.)

Actually, lemme document that.

File: test/test_jit.py [hunk 2]
```diff
@@ -5952,84 +5952,86 @@ def func(a, b):
...
-        def test_log10_int(x):
-            # type: (int) -> float
-            return math.log10(x)
-
-        def test_log10_float(x):
-            # type: (float) -> float
-            return math.log10(x)
+        inf = float("inf")
+        NaN = float("nan")
+        mx_int = 2147483647
```
Comment (driazati) [inline test/test_jit.py]: we use `int64_t` internally so we should try be `2 ** 31`, `2 ** 32`, and `2 ** 63` (and the negations)

Comment (Chillee) [reply test/test_jit.py]: I do the first 2 - I'll add the last one.

File: test/test_jit.py [hunk 3]
```diff
@@ -5952,84 +5952,86 @@ def func(a, b):
...
+        mn_int = -2147483647 - 1
+        float_vals = [inf, NaN, 0.0, 1.0, 2.2, -1.0, -0.0, -2.2, -inf, 1, 0, 2]
```
Comment (driazati) [inline test/test_jit.py]: Test big values for float too

File: test/test_jit.py [hunk 4]
```diff
@@ -5952,84 +5952,86 @@ def func(a, b):
...
+                vals = float_vals if is_float else int_vals
+                vals = [(i, j) for i in vals for j in vals]
 
-        def test_sqrt_float(x):
-            # type: (float) -> float
-            return math.sqrt(x)
+            for a, b in vals:
+                resf = None
+                resfs = None
+                try:
```
Comment (driazati) [inline test/test_jit.py]: This flow is hard to follow (and the names could be more descriptive), it seems like it could mostly be replaced with `self.assertEqual`

File: test/test_jit.py [hunk 5]
```diff
@@ -5952,84 +5952,86 @@ def func(a, b):
...
+                            continue
+                        if isinstance(resf, float) and math.isnan(resf) and math.isnan(resfs):
+                            continue
+                        if isinstance(resf, float) and abs(resf - resfs) < 1e-4:
+                            continue
+                    raise AssertionError("Failed on {func_name} with inputs {a} {b}. Python: {resf}, Script: {resfs}"
+                                         .format(func_name=func_name, a=a, b=b, resf=resf, resfs=resfs))
+
+
+        unary_float_ops = ["log", "log1p", "log10", "exp", "sqrt", "gamma", "lgamma", "erf", "erfc", "expm1", "fabs"]
```
Comment (driazati) [inline test/test_jit.py]: We should have something like `[x for x in dir(math) if callable(getattr(math, x))]` so we know we're getting full coverage of the `math` module instead of maintaining big lists like these (though we still need some list to say what the args are for the non-unary ones)

File: test/test_jit.py (hunk 4 above)
Comment (Chillee) [reply test/test_jit.py]: There's a comment about the conditions that `self.assertEqual` doesn't support - is it possible to make it support that? https://github.com/pytorch/pytorch/pull/21125/files#diff-78aab5b4217aeaed920efab0c5f5f5c8R6009

I suppose it might be able to replace the last if statement or so...

File: test/test_jit.py (hunk 5 above)
Comment (Chillee) [reply test/test_jit.py]: I'd prefer to do this after this stack lol - otherwise I'd need to maintain a list of what functions aren't supported that I whittle down for each commit in this stack.

File: test/test_jit.py (hunk 4 above)
Comment (driazati) [reply test/test_jit.py]: Oh yeah I missed that, the `NaN` thing we shouldn't change so I guess it's not all that useful,  what do you mean by `2. We want to return `nan` for math.sqrt(-5)`, shouldn't it be an exception?

File: test/test_jit.py (hunk 2 above)
Comment (Chillee) [reply test/test_jit.py]: Actually, I don't think it's a good idea. Python has big ints, so it'll invariably have different behavior than what we do near the boundary.

**Review by ailzhang — APPROVED** [review]
LGTM!
